lib/logger/lib/logger/app.ex

```diff
@@ -77,64 +65,64 @@ defmodule Logger.App do
   Stops the application without sending messages to error logger.
   """
   def stop() do
-    try do
-      Logger.Config.deleted_handlers([])
-    catch
-      :exit, {:noproc, _} ->
-        {:error, {:not_started, :logger}}
-    else
-      deleted_handlers ->
-        result = Application.stop(:logger)
-        add_handlers(deleted_handlers)
-        result
-    end
-  end
+    :ok = :logger.update_primary_config(%{level: :none})
 
-  # TODO: Remove conditional error_logger code once we require Erlang/OTP 21+.
-  defp otp_logger? do
-    is_pid(Process.whereis(:logger))
+    Application.stop(:logger)
   end
 
-  defp add_elixir_handler(sasl_reports?) do
-    config = %{level: :debug, config: %{sasl_reports?: sasl_reports?}}
-    :logger.add_handler(Logger, Logger.ErlangHandler, config)
+  defp add_elixir_handler(otp_reports?, counter) do
+    level =
+      case Application.fetch_env!(:logger, :level) do
+        :warn -> :warning
+        other -> other
+      end
+
+    # TODO: This will not be needed once we are OTP 22+
+    data = %{counter: counter}
+
+    config = %{
+      level: :all,
+      config: data,
+      filter_default: :log,
+      filters:
+        if not otp_reports? do
+          [
+            filter_elixir: {&Logger.Filter.elixir_domain/2, :ignore}
+          ]
+        else
+          []
+        end
+    }
+
+    primary_config = :logger.get_primary_config()
+
+    :ok = :logger.add_primary_filter(:process_disabled, {&Logger.Filter.process_disabled/2, []})
```

Comment:
Sorry, I had a slightly change of mind here. I don't want to add a filter to everything yet given Erlang/OTP may or may not add this behaviour, so let's remove the `process_disabled` filter for now and keep it only for Elixir's logger (which we already check on the `log` call).

Reply:
There was a problem with that filter removed, messages that are sent not via `Logger` will be delivered to the backend. For example processes spawned via `proc_lib` that raises message. Also I as a user would found that behaviour a little bit confusing as:

```
Logger.disable()

:logger.info("Foo bar")
```

Will still log messages, which can be counterintuitive, at least from my point of view.

Reply:
Good point. Given that `Logger.configure` changes the level for everyone, it makes sense if `Logger.disable` disables it for everyone too. Thanks!